src/screens/NetworkSettings/CustomNetworkSettingsScreen.js

```diff
@@ -0,0 +1,222 @@
+    }
+
+    setErrorMessages(errors);
+    if (Object.keys(errors).length > 0) {
+      return;
+    }
+
+    dispatch(networkSettingsUpdate(formData));
+  };
+
+  useEffect(() => {
+    setErrorMessages({
+      nodeUrl: networkSettingsErrors?.nodeUrl || '',
+      explorerUrl: networkSettingsErrors?.explorerUrl || '',
+      explorerServiceUrl: networkSettingsErrors?.explorerServiceUrl || '',
+      walletServiceUrl: networkSettingsErrors?.walletServiceUrl || '',
+      walletServiceWsUrl: networkSettingsErrors?.walletServiceWsUrl || '',
+    });
+  }, [networkSettingsErrors]);
+
+  return (
+    <View style={styles.container}>
+      <HathorHeader
+        title={customNetworkSettingsTitleText}
+        onBackPress={() => navigation.goBack()}
+      />
+
+      {isLoading(networkSettingsStatus) && (
+        <FeedbackModal
+          icon={<Spinner />}
+          text={feedbackLoadingText}
+        />
+      )}
+
+      {hasFailed(networkSettingsStatus) && (
+        <FeedbackModal
+          icon={(<Image source={errorIcon} style={styles.feedbackModalIcon} resizeMode='contain' />)}
+          text={feedbackFailedText}
+          onDismiss={handleFeedbackModalDismiss}
+        />
+      )}
+
+      <View style={styles.content}>
+        <View style={styles.warningContainer}>
+          <Text style={styles.warningMessage}>{warningText}</Text>
+        </View>
+        <SimpleInput
+          containerStyle={styles.input}
+          label={t`Node URL`}
+          autoFocus
+          onChangeText={handleInputChange('nodeUrl')}
+          error={errorMessages.nodeUrl}
+          value={formData.nodeUrl}
+        />
+
+        <SimpleInput
+          containerStyle={styles.input}
+          label={t`Explorer URL`}
+          autoFocus
+          onChangeText={handleInputChange('explorerUrl')}
+          error={errorMessages.explorerUrl}
+          value={formData.explorerUrl}
+        />
+
+        <SimpleInput
+          containerStyle={styles.input}
+          label={t`Explorer Service URL`}
+          autoFocus
+          onChangeText={handleInputChange('explorerServiceUrl')}
+          error={errorMessages.explorerServiceUrl}
+          value={formData.explorerServiceUrl}
+        />
+
+        <SimpleInput
+          containerStyle={styles.input}
+          label={t`Wallet Service URL (optional)`}
+          autoFocus
+          onChangeText={handleInputChange('walletServiceUrl')}
+          error={errorMessages.walletServiceUrl}
+          value={formData.walletServiceUrl}
+        />
+
+        <SimpleInput
+          containerStyle={styles.input}
+          label={t`Wallet Service WS URL (optional)`}
+          autoFocus
+          onChangeText={handleInputChange('walletServiceWsUrl')}
+          error={errorMessages.walletServiceWsUrl}
+          value={formData.walletServiceWsUrl}
+        />
+
+        <View style={styles.buttonContainer}>
+          <NewHathorButton
```

Comment:
As a suggestion, we can use the `disabled` attribute here to make it disabled while the required fields are not written. We use it in many forms in the wallet mobile.

Reply:
It was necessary to make a refactor for this to work properly:
- refactor formModel and errorModel: fac7a46e1eded0ca206527a4e8a98e6a29081188
- fix infinity loading: b5b8424c74469bbee1324dfeb7127569d79d7465
- fix dirty errorModel after component unmount: 899e06c97978fbdc65dc125673ef0eb6572cb800